File: src/main/java/responses/Media.java
package responses;

public class Media {
    private final String type;

    public Media(String type) {
        this.type = type;
    }

    public String getType() {
        return type;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Media media = (Media) o;

        return !(type != null ? !type.equals(media.type) : media.type != null);

    }

    @Override
    public int hashCode() {
        return type != null ? type.hashCode() : 0;
    }

    @Override
    public String toString() {
        return "Media{" +
                "type='" + type + '\'' +
                '}';
    }
}

File: src/main/java/com/chrisdempewolf/pinterest/responses/pin/Media.kt
package com.chrisdempewolf.pinterest.responses.pin

class Media(val type: String?) {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val media = o as Media?

        return !if (type != null) type != media!!.type else media!!.type != null

    }

    override fun hashCode(): Int {
        return if (type != null) type.hashCode() else 0
    }

    override fun toString(): String {
        return "Media{" +
                "type='" + type + '\'' +
                '}'
    }
}
